File: solutions/python/secret-handshake/1/secret_handshake.py
def commands(binary_str):
    actions = ["wink", "double blink", "close your eyes", "jump"]
    number_actions = 4
    output = []
    for idx in range(len(binary_str)-1,0,-1):
        if binary_str[idx] == "1":
            output.append(actions[number_actions-idx])
    if binary_str[0]=="1":
            output.reverse()
    return output
            

File: solutions/julia/secret-handshake/1/secret-handshake.jl
function secret_handshake(code)
    actions = ["wink", "double blink", "close your eyes", "jump"]
    output = []
    idx = 1
    bits = reverse(bitstring(UInt16(code)))
    for i in 1:4
        bits[i]=='1' && push!(output, actions[i])
    end
    bits[5] == '1' && return reverse(output)
    output                                
end
